outport/process/transactionsfee/transactionChecker.go

```diff
@@ -0,0 +1,47 @@
+package transactionsfee
+
+import (
+	"bytes"
+	"encoding/hex"
+	"math/big"
+
+	"github.com/ElrondNetwork/elrond-go-core/core"
+	"github.com/ElrondNetwork/elrond-go-core/data"
+	"github.com/ElrondNetwork/elrond-go-core/data/smartContractResult"
+	vmcommon "github.com/ElrondNetwork/elrond-vm-common"
+)
+
+func isSCRForSenderWithRefund(scr *smartContractResult.SmartContractResult, txHash []byte, tx data.TransactionHandlerWithGasUsedAndFee) bool {
+	isForSender := bytes.Equal(scr.RcvAddr, tx.GetSndAddr())
+	isRightNonce := scr.Nonce == tx.GetNonce()+1
+	isFromCurrentTx := bytes.Equal(scr.PrevTxHash, txHash)
+	isScrDataOk := isDataOk(scr.Data)
+
+	return isFromCurrentTx && isForSender && isRightNonce && isScrDataOk
+}
+
+func isRefundForRelayed(dbScResult *smartContractResult.SmartContractResult, tx data.TransactionHandlerWithGasUsedAndFee) bool {
+	isForRelayed := string(dbScResult.ReturnMessage) == core.GasRefundForRelayerMessage
```

Comment:
is there a case when the return message will be `gas refund for relayer` + `@ something`? 
also, what if a user-defined smart contract uses that exact message? will the code be able to differentiate?

Reply:
return message can be set by a smart contract user just in case of error